### ISSUE TITLE
init: --subdir is not being enforced

### ISSUE THREAD

**Issue by jorgeorpinel** [iterative/dvc#4302]
## Bug Report

```dvc
$ mkdir test && cd test
$ git init && dvc init
$ mkdir dir && cd dir
$ git init && dvc init
  # No problem

$ cd ..
$ dvc destroy -f
$ dvc init
  # No problem
```

- Also, should it also be enforced even with `dvc init --no-scm` ? (I forgot the original motive of `--subdir`).

### Please provide information about your setup

**Output of `dvc version`:**

```console
$ dvc version
DVC version: 1.2.1+b14ad4.mod
Python version: 3.8.2
Platform: Windows-10-10.0.18362-SP0
Binary: False
Package: None
Supported remotes: azure, gdrive, gs, hdfs, http, https, s3, ssh, oss
Repo: dvc, git
Filesystem type (workspace): ('NTFS', 'C:\\')
```

**Additional Information (if any):**

If applicable, please also provide a `--verbose` output of the command, eg: `dvc add --verbose`.

**Comment by jorgeorpinel** [comment]
Cc @pared

**Comment by efiop** [comment]
> $ git init && dvc init
>   # No problem (expecting ERROR)

@jorgeorpinel This first one won't cause an error, because git is separate, and --subdir is not needed.

**Comment by jorgeorpinel** [comment]
You're right. I guess it's a matter of improving the explanation of all this in docs then. From reading it I didn't remember the behavior correctly. Transferring!
